src/rgw/rgw_rados.cc

```diff
@@ -3764,6 +3764,22 @@ int RGWRados::init_rados()
     }
   }
 
+  {
```

Comment:
@liewegas @mattbenjamin yeah, I'd go with (1), and maybe have some mechanism to explicitly set (2) when needed (e.g.,  when running `radosgw-admin metadata sync run`).